lib/components/viewers/related-stops-panel.js

```diff
@@ -0,0 +1,107 @@
+import React, { Component } from 'react'
+
+import {
+  getModeFromRoute,
+  getRouteIdForPattern,
+  getStopName,
+  getStopTimesByPattern,
+  routeIsValid
+} from '../../util/viewer'
+import Icon from '../narrative/mode-icon'
+import { ComponentContext } from '../../util/contexts'
+
+import NextArrivalForPattern from './next-arrival-for-pattern'
+import RelatedPanel from './related-panel'
+import ViewStopButton from './view-stop-button'
+
+class RelatedStopsPanel extends Component {
+  constructor (props) {
+    super(props)
+    this.state = {
+      expanded: false
+    }
+  }
+
+  static contextType = ComponentContext
+
+  _toggleExpandedView = () => {
+    this.setState({ expanded: !this.state.expanded })
+  }
+
+  _getStopIcon = (stop) => {
+    const { ModeIcon } = this.context
+    const modes = [...new Set(stop.routes?.map(getModeFromRoute))]
+    if (modes.length === 1) {
+      return <ModeIcon height={22} mode={modes[0]} width={22} />
+    }
+    return <Icon type='map-marker' />
+  }
+
+  render () {
+    const { expanded } = this.state
+    const {
+      homeTimezone,
+      nearbyStops,
+      stopViewerArriving,
+      stopViewerConfig,
+      timeFormat
+    } = this.props
+    return (
+      <RelatedPanel
+        count={nearbyStops?.length}
+        expanded={expanded}
+        onToggleExpanded={this._toggleExpandedView}
+        title='Related Stops'
+      >
+        <ul className='related-items-list list-unstyled'>
+          {nearbyStops?.map((stop, index) => {
+            const { id } = stop
+            if (!expanded && index >= 2) return null
+            const stopTimesByPattern = getStopTimesByPattern(stop)
+            const label = getStopName(stop)
+
+            return (
+              <li className='related-item' key={id}>
+                <div className='stop-label'>
+                  <div className='stop-name overflow-ellipsis' title={label}>
+                    <span className='child-stop-icon'>
+                      {this._getStopIcon(stop)}
+                    </span>
+                    <span className='stop-name-text'>{label}</span>
+                  </div>
+                  <ViewStopButton
+                    className='view-child-stop-button'
+                    stopId={stop.id}
+                    text='View details'
```

Comment:
Clicking this link should remove the a popover from the map, if one is shown. Otherwise, the map highlight isn't as useful, since it is not as large as the popover. In this photo, the red circled stop should have the user's attention, but it's very much the pink circled stop that does.
<img width="733" alt="Screen Shot 2021-09-14 at 11 55 14 AM" src="https://user-images.githubusercontent.com/86619099/133245359-4a96248c-fed4-4e22-b75a-66fa1d9a9032.png">

Reply:
Looks like we'll have to table this one for when we refactor the ugly bubbles. Looks like a lot of otp-ui tie-in.